doc/source/HOW-TO/cluster-configuration.rst

```diff
@@ -1,38 +1,72 @@
 
-Install master and worker nodes
-===============================
+Configure master and worker nodes
+=================================
 
-On each node:
+1. Set up environment on master and all workers:
 
-- setup qserv
-- make fresh run directory
-- edit meta-configuration file
-- fill run directory.
+.. code-block:: bash
 
-Example for master node:
+  source <path to stack>/loadLSST.bash
+  cd <path to respositories>
+  setup -r qserv
+  setup -r qserv_testdata -k
 
-.. code-block:: bash
+2. Configure Qserv on master and all workers:
+
+- Run :code:`qserv-configure.py -i -R <qserv-run directory>`
+
+- Edit :code:`<qserv-run directory>/qserv-meta.conf`
+ 
+  - Change :code:`node_type` to :code:`master` or :code:`worker`
+  - | Change master DNS name to the name of the machine as shown by :code:`hostname -s`.
+    | Do not alter it in any way, for example, by adding the full domain name.
+    | On the worker, use the master's DNS name, **not** the worker's
+
+- Rerun :code:`qserv-configure` without :code:`-i` as :code:`qserv-configure.py -R <qserv-run directory>`
+
+3. Copy secret file from the master to every worker:
+
+   *Note: secret file is read when qserv-wmgr client starts*
+
+   :code:`scp <qserv-run directory>/etc/wmgr.secret> <worker1 hostName>:<qserv-run directory>/etc/wmgr.secret`
+   :code:`scp <qserv-run directory>/etc/wmgr.secret> <worker2 hostName>:<qserv-run directory>/etc/wmgr.secret`
+
+4. Start services on master and every worker:
 
-  . /path/to/stack/loadLSST.bash
-  setup qserv $VERSION
-  qserv-configure.py -i -R /path/to/run/dir -D /path/to/data/dir
-  vi /path/to/run/dir/qserv-meta.conf
-  # change values of next: 
-  #   - "node_type" is set to "master"
-  #   - "master" is set to master dns name
+   :code:`<qserv-run directory>/bin/qserv-start.sh`
 
-  # next command will not impact /path/to/data/dir if not empty
-  qserv-configure.py -r /path/to/run/dir
-  /path/to/run/dir/bin/qserv-start.sh
+5. Grant permissions on every worker:
 
-And then repeat the very same operations on the worker nodes,
-except that "node_type" field value is set to "worker".
+   :code:`mysql -S <qserv-run directory>/var/lib/mysql/mysql.sock -uroot -p`
+
+   .. code-block::
+
+     GRANT USAGE ON *.* TO 'qsmaster'@'<master hostName>';
+     GRANT SELECT ON `mysql`.* TO 'qsmaster'@'<master hostName>';
+     GRANT ALL PRIVILEGES ON `qservTest%`.* TO 'qsmaster'@'<master hostName>';
+     GRANT ALL PRIVILEGES ON `qservw_worker`.* TO 'qsmaster'@'<master hostName>';
+
+6. On the master, register workers with CSS:
+
+   :code:`qserv-admin.py`
+
+   .. code-block::
+
+     CREATE NODE worker1 type=worker port=5012 host=<worker1 hostName>;
+     CREATE NODE worker2 type=worker port=5012 host=<worker2 hostName>;
+
+Run Integration Test
+====================
+
+.. code-block:: bash
 
+  qserv-check-integration.py --case=01 --load -M --mode=qserv
+  qserv-check-integration.py --case=02 --load -M --mode=qserv
```

Comment:
I think "..." will be better in case we have more than 5 test cases in the future.